src/components/Card.js

```diff
@@ -16,39 +35,76 @@ export default class Card {
     this._cardElement
       .querySelector(".card__like-button")
       .addEventListener("click", () => {
-        this._handleLikeIcon();
+        this._handleLikeClick(this);
       });
 
     // Delete button
-    this._cardElement
-      .querySelector(".card__delete-button")
-      .addEventListener("click", () => {
-        this._handleDeleteCard();
-      });
+    if (this.myID === this._ownerId) {
+      this._cardElement
+        .querySelector(".card__delete-button")
+        .addEventListener("click", () => {
+          this._handleDeleteCardPopup(this);
+        });
+    }
 
+    // Click on image for preview
     this._cardElement
       .querySelector(".card__image")
       .addEventListener("click", () => {
         this._handleCardClick(this._getData());
       });
   };
 
-  _handleLikeIcon() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
-  }
-
-  _handleDeleteCard() {
+  handleDeleteCard() {
     this._cardElement.remove();
     this._cardElement = null;
   }
 
+  isLiked() {
+    return this._likes.some((like) => like._id === this.myID);
+  }
+
   _fillCardTemplate() {
     const imageElement = this._cardElement.querySelector(".card__image");
     imageElement.src = this.link;
     imageElement.alt = this.name;
+    this._cardElement.id = this.imageID;
+
     this._cardElement.querySelector(".card__caption").textContent = this.name;
+    if (this.myID !== this._ownerId) {
+      this._cardElement.querySelector(".card__delete-button").remove();
+    }
+
+    if (this.isLiked()) {
```

Comment:
You should render the likes without conditions here, since there can be likes even if you didn't like the card

Reply:
.